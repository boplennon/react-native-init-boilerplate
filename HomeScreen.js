import * as React from 'react';
import styles from './styles';
import { Text, View } from "react-native";
import { FormLabel, Icon } from 'react-native-elements';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export const HomeScreen = () => (
    <View style={styles.container}>
        <Text style={styles.welcome}>React Native "native" SanityCheck</Text>
        <Text style={styles.instructions}>Tested Libraries:</Text>
        <View>
            <FormLabel>- react-native-vector-icons</FormLabel>
            <FormLabel>- react-native-elements</FormLabel>
            <FormLabel>- react-apollo</FormLabel>
            <FormLabel>- graphql</FormLabel>
            <FormLabel>- apollo-boost</FormLabel>
            <FormLabel>- aws-amplify</FormLabel>
        </View>
        <FormLabel>React-Native Elements with icons</FormLabel>
        <FontAwesome name='map-o' color='blue' size={48} />
        <Icon type='font-awesome' color='purple' name='user-o' size={48} />
    </View>);

export default HomeScreen;