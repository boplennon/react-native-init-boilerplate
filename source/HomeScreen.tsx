import * as React from 'react';
import styles from './styles';
import { Text, View, ScrollView } from "react-native";
import { FormLabel, Icon } from 'react-native-elements';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

export const HomeScreen = () => (
    <View style={styles.container}>
        <Text style={styles.welcome}>React Native "native" SanityCheck</Text>
        <Text style={styles.instructions}>Tested Libraries:</Text>
        <ScrollView>
        <View>
            <FormLabel>- 16.5.0 react</FormLabel>
            <FormLabel>- 0.57.0 react-native</FormLabel>
            <FormLabel>- 3.0.3 typescript</FormLabel>
            <FormLabel>- 23.1.4 ts-jest</FormLabel>
            <FormLabel>- 2.14.2 react-navigation</FormLabel>
            <FormLabel>- 5.0.0 react-native-vector-icons</FormLabel>
            <FormLabel>- 0.19.0 react-native-elements</FormLabel>
            <FormLabel>- 2.1.11 react-apollo</FormLabel>
            <FormLabel>- 14.0.2 graphql</FormLabel>
            <FormLabel>- 0.1.16 apollo-boost</FormLabel>
            <FormLabel>- 1.1.1 aws-amplify</FormLabel>
        </View>
        </ScrollView>
        <FormLabel>React-Native Elements with icons</FormLabel>
        <FontAwesome name='map-o' color='blue' size={48} />
        <Icon type='font-awesome' color='purple' name='user-o' size={48} />
    </View>);

export default HomeScreen;