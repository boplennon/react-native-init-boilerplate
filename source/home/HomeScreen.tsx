import * as React from 'react';
import { ScrollView, Text, View } from 'react-native';
import { FormLabel, Icon } from 'react-native-elements';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import { CrossTabStatelessComponent } from '../ReactNavigator';
import styles, { TabIconSize } from '../styles';

export const HomeScreen: CrossTabStatelessComponent = () => (
  <View style={styles.container}>
    <Text>React Native "native" SanityCheck</Text>
    <Text>Tested Libraries:</Text>
    <ScrollView>
      <View>
        <FormLabel>- 16.5.0 react</FormLabel>
        <FormLabel>- 0.57.0 react-native</FormLabel>
        <FormLabel>- 3.0.3 typescript</FormLabel>
        <FormLabel>- 23.1.4 ts-jest</FormLabel>
        <FormLabel>- 0.39.0 react-native-sentry</FormLabel>
        <FormLabel>- 0.21.0 react-native-maps</FormLabel>
        <FormLabel>- 2.14.2 react-navigation</FormLabel>
        <FormLabel>- ^2.0.1 react-native-paper</FormLabel>
        <FormLabel>- ^0.4.0 react-navigation-material-bottom-tabs</FormLabel>
        <FormLabel>- 5.0.0 react-native-vector-icons</FormLabel>
        <FormLabel>- 0.19.0 react-native-elements</FormLabel>
        <FormLabel>- 2.1.11 react-apollo</FormLabel>
        <FormLabel>- 14.0.2 graphql</FormLabel>
        <FormLabel>- 0.1.16 apollo-boost</FormLabel>
        <FormLabel>- 1.1.1 aws-amplify</FormLabel>
      </View>
    </ScrollView>
    <FormLabel>Basic icon and one from Elements</FormLabel>
    <FontAwesome name='map-o' color='blue' size={48} />
    <Icon type='font-awesome' color='purple' name='user-o' size={48} />
  </View>
);

HomeScreen.navigationOptions = {
  tabBarIcon: <FontAwesome name='home' size={TabIconSize} color='white' />,
  tabBarLabel: <Text style={styles.tabBarLabel}>Home</Text>,
};

export default HomeScreen;
