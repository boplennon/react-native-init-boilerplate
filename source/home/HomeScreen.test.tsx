import React from 'react';
import HomeScreen from './HomeScreen';
import TestRenderer from 'react-test-renderer';

function setup() {
  return TestRenderer.create(
    <HomeScreen />
  );
}

describe('components', () => {
  describe('<HomeScreen />', () => {
    it('renders correctly with defaults', () => {
      const wrapper = setup();
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
