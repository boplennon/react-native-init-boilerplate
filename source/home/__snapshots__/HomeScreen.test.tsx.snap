// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <HomeScreen /> renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          },
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          React Native Init Boilerplate
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          Tested Libraries:
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          - 16.5.0 react
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View>
          - 0.57.0 react-native
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <View
          onPress={[Function]}
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
